src/main/java/org/jabref/logic/autosaveandbackup/AutosaveManager.java

```diff
@@ -21,38 +16,40 @@
 
 /**
  * Saves the given {@link BibDatabaseContext} on every {@link BibDatabaseContextChangedEvent} by posting a new {@link AutosaveEvent}.
- * An intelligent {@link ExecutorService} with a {@link BlockingQueue} prevents a high load while saving and rejects all redundant save tasks.
+ * An intelligent {@link ScheduledThreadPoolExecutor} prevents a high load while saving and rejects all redundant save tasks.
+ * The scheduled action is stored and canceled if a newer save action is proposed.
  */
 public class AutosaveManager {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AutosaveManager.class);
+    private static final int AUTO_SAVE_DELAY = 200;
 
     private static Set<AutosaveManager> runningInstances = new HashSet<>();
 
     private final BibDatabaseContext bibDatabaseContext;
-    private final BlockingQueue<Runnable> workerQueue;
-    private final ExecutorService executor;
+    private final ScheduledExecutorService executor;
     private final EventBus eventBus;
     private final CoarseChangeFilter changeFilter;
+    private Future<?> scheduledSaveAction;
```

Comment:
Can `?` be replaced by `Void` as no return value is used?

Reply:
I would need to do a cast then on line 53.